tests/basics/async_coroutine.py

```diff
@@ -0,0 +1,13 @@
+async def f():
+    pass
+
+try:
+    f()  # Should not crash
+except Exception as e:
+    print('failed to invoke')
+
+try:
+    next(f())
+    print('This should fail because async def returns a coroutine, and next() is not allowed')
+except Exception as e:
```

Comment:
Want to make this TypeError explicitly?

Reply:
I do, but if you catch TypeError only, the test run that does
```
MICROPY_CPYTHON3=python3.8 MICROPY_MICROPYTHON=../ports/unix/micropython_coverage ./run-tests -j1 --via-mpy -d basics float
```
fails.